sklearn/model_selection/tests/test_split.py

```diff
@@ -216,6 +216,7 @@ def test_kfold_valueerrors():
     # though all the classes are not necessarily represented at on each
     # side of the split at each split
     with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
```

Comment:
Not related to this PR, but what is the rationale behind not raising an error for this extreme case when it creates an empty test fold. ie the number of labels for all classes is less than the number of folds?

It is highly likely that this will raise an error at a further stage. For ex

    dtc = DecisionTreeClassifier()
    X2 = np.array([[1, 2], [3, 4], [5, 6], [7, 8], [9, 10]])
    y = np.array([3, 3, -1, -1, 2])
    cross_val_score(dtc, X2, y)